core/src/main/java/io/micronaut/core/util/ArrayUtils.java

```diff
@@ -172,6 +175,18 @@ public static String toString(String delimiter, @Nullable Object[] array) {
         return collection.toArray(array);
     }
 
+    /**
+     * Returns an array containing all of the elements in this collection, using the item class.
+     *
+     * @param collection The collection
+     * @param arrayItemClass The array item class
+     * @param <T> The type of the array
+     * @return The array
+     */
+    public static <T> T[] toArray(Collection<T> collection, Class<T> arrayItemClass) {
```

Comment:
Add `@UsedByGeneratedCode`

Reply:
Actually, I end up using generated `.toArray` instead of this method. I can delete it or I will just add `@since`